raiden_contracts/data/source/services/MonitoringService.sol

```diff
@@ -177,8 +177,13 @@ contract MonitoringService is Utils {
     )
         public
     {
-        // Call updateReward first to avoid reentrancy problems when the
-        // token_network_address belongs to a malicious contract.
+        // Here we're trying to do bookkeeping first, but updateReward() first calls
+        // token_network_address.  So reentrancy is possible.
```

Comment:
I see that it calls `TokenNetwork` on `token_network_address`.

Reply:
TokenNetwork(...) is like a mypy `BlockNumber(...)`.  It doesn't check anything at runtime.